### PR TITLE
HD-725 add new liberasurecode & use new encode

### PR DESCRIPTION
**Pull request by benoit-a**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
 
     - name: Build liberasurecode
       run: |
-        git clone https://github.com/openstack/liberasurecode.git
```
Comment (benoit-a) [inline .github/workflows/go.yml:36]: to remove

File: backend.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package erasurecode
 
 /*
+#cgo CFLAGS: -fno-omit-frame-pointer
```
Comment (benoit-a) [inline backend.go:4]: to remove

File: backend.go [hunk 3]
```diff
@@ -357,8 +358,21 @@ func BackendIsAvailable(name string) bool {
...
+	return initBackend(params, C.CHKSUM_NONE)
+}
+
 // InitBackend returns a backend descriptor according params provided
 func InitBackend(params Params) (Backend, error) {
+	return initBackend(params, C.CHKSUM_CRC32)
+}
+
+func InitBackendXH(params Params) (Backend, error) {
+	return initBackend(params, C.CHKSUM_XXHASH)
```
Comment (benoit-a) [inline backend.go:372]: Could it be part of the "Params" ?

File: backend_test.go [hunk 4]
```diff
@@ -557,12 +557,13 @@ func (d decodeTest) String() string {
...
 	{1024 * 1024, Params{Name: "isa_l_rs_vand", K: 4, M: 2, W: 8, HD: 5}},
 	{5 * 100000, Params{Name: "isa_l_rs_vand", K: 5, M: 7}},
-	// Will force an allocation of a new (dedicated) pool
-	{10000000, Params{Name: "isa_l_rs_vand", K: 2, M: 1, MaxBlockSize: 10000000 + maxBuffer}},
-	// Will force an allocation at every encoding (but should work)
-	{maxBuffer * 2, Params{Name: "isa_l_rs_vand", K: 2, M: 1, MaxBlockSize: maxBuffer / 2}},
+	// 	// Will force an allocation of a new (dedicated) pool
+	// 	{10000000, Params{Name: "isa_l_rs_vand", K: 2, M: 1, MaxBlockSize: 10000000 + maxBuffer}},
+	// 	// Will force an allocation at every encoding (but should work)
+	// 	{maxBuffer * 2, Params{Name: "isa_l_rs_vand", K: 2, M: 1, MaxBlockSize: maxBuffer / 2}},
```
Comment (benoit-a) [inline backend_test.go:566]: to reactivate

File: backend_test.go [hunk 5]
```diff
@@ -722,20 +723,73 @@ func BenchmarkDecodeMSlow(b *testing.B) {
 }
 
 func BenchmarkEncodeM(b *testing.B) {
```
Comment (benoit-a) [inline backend_test.go:725]: Cleanup / remove ? Maybe a dedicated bench for checksum ?

**Review by fferrandis — APPROVED** [review]

**Review by slaperche-scality — APPROVED** [review]

**Pull request merged**